nad_ch/controllers/cli.py

```diff
@@ -11,28 +15,33 @@ def cli(ctx):
 
 @cli.command()
 @click.pass_context
-@click.argument('filename')
-@click.argument('content')
-def upload(ctx, filename, content):
+@click.argument('provider_name')
+def add_provider(ctx, provider_name):
     context = ctx.obj
-    file = File(name=filename, content=content)
-    upload_file(context, file)
-    click.echo(f"Uploaded {filename}")
+    try:
+        add_data_provider(context, provider_name)
+    except InvalidProviderNameException as e:
+        click.echo(f"Error: {e.message}")
+        return
+
+    click.echo('Provider added')
 
 
 @cli.command()
 @click.pass_context
-def listall(ctx):
+def list_providers(ctx):
     context = ctx.obj
-    files = list_files(context)
-    for file in files:
-        click.echo(file.name)
+    providers = list_data_providers(context)
+    click.echo('Data Provider Names:')
+    for p in providers:
+        click.echo(p.name)
```

Comment:
I'd consider putting the printing in `list_data_providers`, and maybe use a logging interface via the app context to handle the actual output.

Reply:
Agree that using a logging interface via the app context makes sense, see below for more thoughts.